src/ucp/tag/tag_recv.c

```diff
@@ -157,15 +193,17 @@ ucp_tag_recv_common(ucp_worker_h worker, void *buffer, size_t buffer_size,
     } else if (save_rreq) {
         /* If not found on unexpected, wait until it arrives.
          * If was found but need this receive request for later completion, save it */
+        context            = worker->context;
+        queue              = ucp_tag_exp_get_queue(&context->tm, tag, tag_mask);
         req->recv.buffer   = buffer;
         req->recv.length   = buffer_size;
         req->recv.datatype = datatype;
         req->recv.tag      = tag;
         req->recv.tag_mask = tag_mask;
         req->recv.cb       = cb;
-        ucp_tag_exp_add(&worker->context->tm, req);
-        ucs_trace_req("%s returning expected request %p (%p)", debug_name,
-                      req, req + 1);
+        ucp_tag_exp_push(&context->tm, queue, req);
```

Comment:
just curious: does it provide performance benefit over the old func with request?